### ISSUE TITLE
Introduction(BACKEND) - Social Media 

### ISSUE THREAD

**Issue by anslemkelechi** [Cirle-7/social-media-app#1]
# Overview <html>
<body>
<b style="font-weight:normal;" id="docs-internal-guid-fd3789c3-7fff-cfb7-a65e-15956f7758cc"><div dir="ltr" style="margin-left:0pt;" align="left">

We’re building a social media application in the likeness of twitter but for developers in mind.
--

</div></b>
</body>
</html>

# Stack 
*Language : Javascript(nodejs)*
*Database: PostgreSql, MySql*

# Basic Features 
- Publish a new post
- Like a post
- Comment on a post
- Register a new account(JWT & social authentication)
- Login to account (JWT & social authentication)
- Share user profile
- Upload images and videos to posts
- Follow other users on the platform
- Share/repost a post
- Edit& Delete  a post
- Delete account
- View all users 
- Block all users
- Unblock users
- Delete user
- Unpublish post
- Messaging & rooms

# Tasks 
- [x] #2
- [x] Database Setup
- [x] #10
- [x] #12
- [x] #13

## POST LOGIC
- [x] #19
- [x] #20
- [x] #21
- [x] #37
- [x] #35
- [x] #42
- [x] #38
- [x] #46
- [x] #47
- [x] #39
- [x] #40
- [x] Get All Posts By Topic
- [x] Get All Posts By Location

## USER LOGIC
- [x] #25
- [x] #27
- [x] #28
- [x] #34
- [x] #44
- [ ] #43
- [x] #29
- [ ] Oauth For Third-Party Integration.

*Note: When you create an issue, always tag this issue, also some tasks here can be broken down into subtasks, ensure you create issues for tasks you feel are too large to be compressed into one or can stand on its own. For example : * creation of data models can have its own issue and then multiple sub issues for each data model created, that way multiple individuals can work on a data model per time and we can track major code changes easily*
